utils/config_validator.py

```
@@ -0,0 +1,191 @@
+
+    # Validate log_level
+    if 'log_level' in config:
+        valid_levels = ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL']
+        log_level = config['log_level']
+        if not isinstance(log_level, str):
+            errors.append(f"'log_level' must be a string, got: {type(log_level).__name__}")
+        elif log_level.upper() not in valid_levels:
+            errors.append(f"'log_level' must be one of {valid_levels}, got: '{log_level}'")
+
+    # Validate log_file is a string if provided
+    if 'log_file' in config and config['log_file']:
```

Ruff (0.14.3)

Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)